internal/common/physics/engine.go

```diff
@@ -1,17 +1,10 @@
 package physics
 
 import (
-	//"fmt"
 	utils "SOMAS2023/internal/common/utils"
 	"math"
-	//baseAgent "github.com/MattSScott/basePlatformSOMAS/BaseAgent"
-	//"github.com/google/uuid"
 )
 
-// since in go you cannot define methods on types from other packages
-type Coordinates utils.Coordinates
-type Forces utils.Forces
-
 /*
 The Engine struct is responsible for calculating physics for the environment
```

Comment:
I am wondering about the significance of having an Engine struct since all these functions will have to be available to the environment object (which I guess would need the engine struct this way). 
I think it would make sense to have an IEngine interface if we were thinking of potentially trying out different implementations of the core functions and that way which type of engine is used can be decided when instantiating environment objects. But I don't think we will want to do this so maybe it might be easier to just have these functions available (unless I'm missing something)

Reply:
I don't think we need mulitiple Engine implementations - this should be fixed pretty early on. So what I was thinking for the Server is having one Engine object and then calling things like:

for bike in multibikes:
  for biker in bikers:
    engine.AddAgentForce = biker.DecideForce()
    engine.CalcAcceleration(engine.CalculateForce(), bike.Mass)...

Reply:
Mh I think the significance of having an Environment Object interface is to abstract what you do in the server so that you don't have to call CalcAcceleration, CalcVelocity, GetNewLoc ecc ecc every time you want to move an object (MegaBike or Audi). Instead the Environment Objects will implement some Move() function that will use the engine functions. I think this just makes it a bit neater and keeps the server at a higher level. So for example instead of 
{for bike in multibikes:
for biker in bikers:
      bike.AddAgentForce = biker.DecideForce()
engine.CalcAcceleration(engine.CalculateForce(), bike.Mass)...}
engine.CalcAceleration(engine.CalcForce(), audi.Mass)...

you'd have

{for bike in multibikes: 
for biker in bikers: 
    add force ...
bike.Move()}
audi.Move